src/modules/core/models/filters/filters.model.ts

```diff
@@ -0,0 +1,103 @@
+import { FormGroup } from '@angular/forms';
+import { FilterHandler } from './handlers/base-filter.handler';
+import { CheckboxGroupHandler } from './handlers/checkbox-group.handler';
+import { FilterHandlerFactory } from './handlers/filter-handler.factory';
+
+export type Filter = BaseFilter & (CheckboxGroupFilter | CheckboxesFilter | DateRangeFilter);
+type BaseFilter = { key: string } & FilterOptions;
+
+type CheckboxesFilter = {
+  type: 'CHECKBOXES';
+  checkboxes: ({ key: string; title: string; defaultValue?: boolean } & FilterOptions)[];
+  selected?: { key: string; value: boolean }[];
+};
+
+type CheckboxGroupFilter = { type: 'CHECKBOX_GROUP'; selected?: { key: string; value: string }[] } & CollapsibleFilter;
+
+type DateRangeFilter = {
+  type: 'DATE_RANGE';
+  startDate: DateFilter;
+  endDate: DateFilter;
+  selected?: { key: string; value: string }[];
+} & CollapsibleFilter;
+
+type CollapsibleFilter = { title: string; description?: string; state: 'opened' | 'closed'; scrollable?: boolean };
+type DateFilter = { key: string; label: string; description?: string; defaultValue?: string };
+type FilterOptions = { options?: { updateOn?: 'blur' | 'change'; emitEvent?: boolean } };
+
+type Dataset = { value: string; label: string; description?: string }[];
+
+/**
+ * TODOS:
+ * Pre-load form with values (e.g., from localStorage)
+ * Handle checkbox group search by dataset (i.e., searchable to collapsible)
+ */
+export class FiltersModel {
+  form: FormGroup;
+
+  filters: Filter[];
+  handlers: Map<string, FilterHandler>;
+
+  #datasets: Map<string, Dataset>;
+
+  constructor(config?: Filter[]) {
+    this.form = new FormGroup({}, { updateOn: 'blur' });
+
+    this.filters = [];
+    this.handlers = new Map<string, FilterHandler>();
+
+    this.#datasets = new Map();
+
+    if (config) {
+      for (const filter of config) {
+        this.addFilter(filter);
+      }
+    }
+  }
+
+  addFilter(filter: Filter) {
+    this.filters.push(filter);
+
+    const handler = FilterHandlerFactory.create(filter, this.form);
```

Comment:
This feel like a halfway-house of decoupling the dependencies.  If the model depends on the factory, and the factory is just a switch statement, then maybe the that switch statement should stay in a method in the FiltersModel class OR the handler factory be passed into the constructor.

Reply:
You are right, leaving it for now as is (left a comment).